### PR TITLE
interface: restructured outer directories into views and logic

### PR DESCRIPTION
**Pull request by tacryt-socryp**
The outer src directory was getting extraordinarily messy with the combination of the SPA transition, the global store, and the move over to TypeScript. Reorganized into the following outer structure:
- views
  - components
  - apps
  - css
  - themes
  - app.js
- logic
  - api
  - lib
  - reducers
  - store
  - subscription
- types
- index.js

No functionality has been changed. It builds and renders properly. Internally within these folders I think we could still use some more restructuring, but I think this is a good step forward.

### CODE REVIEW

File: pkg/interface/src/views/apps/chat/components/lib/chat-header.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Link } from "react-router-dom";
 
 import { ChatTabBar } from "./chat-tabbar";
 import { SidebarSwitcher } from "../../../../components/SidebarSwitch";
-import { deSig } from "../../../../lib/util";
+import { deSig } from "../../../../../logic/lib/util";
```
Comment (tylershuster) [inline pkg/interface/src/views/apps/chat/components/lib/chat-header.js:6]: this is a bit deep in'it? maybe we should do something like this https://github.com/entwicklerstube/babel-plugin-root-import

Comment (tacryt-socryp) [reply pkg/interface/src/views/apps/chat/components/lib/chat-header.js:6]: That would be a great quality of life change

Comment (tylershuster) [reply pkg/interface/src/views/apps/chat/components/lib/chat-header.js:6]: I'll PR after this gets merged

**Comment by matildepark** [comment]
I've honestly never heard of a 'logic' folder before, but I haven't exactly looked at a lot of codebases.

**Pull request merged**